packages/next/build/webpack/plugins/next-esm-plugin.ts

```diff
@@ -224,8 +228,27 @@ export default class NextEsmPlugin implements Plugin {
     }
 
     Object.keys(compilerEntries).forEach(entry => {
-      const entryFiles = compilerEntries[entry]
+      const entryFiles = compilerEntries[entry] as (string[] | string)
+
       if (Array.isArray(entryFiles)) {
+        const polyfillIndex = entryFiles.findIndex(file => {
+          return (
+            path.resolve(childCompiler.context, file) ===
+            path.resolve(
+              childCompiler.context,
+              NEXT_PROJECT_ROOT_DIST_CLIENT,
+              'polyfills.js'
+            )
+          )
+        })
+        if (polyfillIndex !== -1) {
```

Comment:
IMO the polyfills will only be needed for nomodule browsers, so do we still need a modern version of `polyfills.js`?